sample_files/main.py

```diff
@@ -13,43 +16,68 @@
 
 # Initialize logging
 logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
 
 # Load environment variables
-from dotenv import load_dotenv
-import os
-
 load_dotenv()
 
 # Fetch API key
 api_key = os.getenv('OPENAI_API_KEY')
 
+if not api_key:
+    logger.error("OPENAI_API_KEY environment variable not found")
+    raise EnvironmentError("OPENAI_API_KEY environment variable not found")
+
 # Initialize OpenAI Chat model
 llm = ChatOpenAI(temperature=0.8, api_key=api_key)
 
-# Initialize Chrome WebDriver using WebDriver Manager
 def get_webdriver():
-    return webdriver.Chrome(ChromeDriverManager().install())
+    """
+    Initialize Chrome WebDriver using WebDriver Manager.
+    """
+    try:
+        return webdriver.Chrome(ChromeDriverManager().install())
+    except WebDriverException as e:
+        logger.error(f"Failed to initialize WebDriver: {e}")
+        raise
 
-# Function to check interactions with Selenium WebDriver
 def check_chain_interactions(chain_results, urls):
+    """
+    Check interactions with Selenium WebDriver.
+
+    Args:
+        chain_results (list): Results from the chain execution.
+        urls (list): List of URLs to check.
+
+    Returns:
+        bool: True if all interactions are successful, False otherwise.
+    """
     for url in urls:
         try:
             driver = get_webdriver()
             driver.get(url)
             for result in chain_results:
-                element = driver.find_element(By.XPATH, f"//*[contains(text(), '{result}')]")
-                if not element:
+                try:
+                    element = driver.find_element(By.XPATH, f"//*[contains(text(), '{result}')]")
+                except NoSuchElementException:
                     logger.error(f"Element {result} not found on the webpage {url}")
                     driver.quit()
                     return False
             driver.quit()
-            return True
         except WebDriverException as e:
             logger.error(f"Selenium WebDriver error: {e}")
             return False
+    return True
 
-# Function to execute the chain process
 def execute_chain(urls, max_attempts=5, attempt=1):
+    """
+    Execute the chain process with retries.
+
+    Args:
+        urls (list): List of URLs to process.
+        max_attempts (int): Maximum number of attempts.
+        attempt (int): Current attempt number.
+    """
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

LGTM!


<!-- This is an auto-generated comment by OSS CodeRabbit -->